include/libcyphal/application/registry/register.hpp

```diff
@@ -173,6 +218,43 @@ class IRegister : public cavl::Node<IRegister>  // NOSONAR cpp:S4963
 
 };  // IRegister
 
+/// Defines type-erased register.
+///
+/// The Footprint size of the unbounded variant should be enough for any register implementation.
+/// The implementation should not be copyable but moveable.
+///
+template <std::size_t Footprint>
+using Register = ImplementationCell<IRegister, cetl::unbounded_variant<Footprint, false, true>>;
```

Comment:
This is a bit late in the game, but do you think that we've made the Libcyphal code more complicated by implementing type erasure using the ImplementationCell? In other words, should we have just used Templates since we're using them elsewhere? I think maybe this is something we could talk about in the dev call

Reply:
Nevermind, Scott allayed my concerns :)